src/components/MyPortfolio/MyPortfolio.tsx

```diff
@@ -0,0 +1,161 @@
+import { ALL_CHAINS } from '@pangolindex/sdk';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import Scrollbars from 'react-custom-scrollbars';
+import { Lock } from 'react-feather';
+import { ThemeContext } from 'styled-components';
+import { usePangolinWeb3 } from 'src/hooks';
+import { PairDataUser, TokenDataUser, useGetChainsBalances, useGetWalletChainTokens } from 'src/state/pportfolio/hooks';
+import { Box } from '../Box';
+import { Loader } from '../Loader';
+import ToggleBalanceButton from '../Portfolio/ToogleBalanceButton';
+import { Text } from '../Text';
+import PortfolioRow from './PortfolioRow';
+import { Body, Frame, Header, Root, SelectedCard } from './styleds';
+
+const MyPortfolio: React.FC = () => {
+  const theme = useContext(ThemeContext);
+  const { account } = usePangolinWeb3();
+
+  const [availableBalances, setAvailableBalances] = useState<{ chainID: number; balance: number }[]>([]);
+  const [showBalances, setShowBalances] = useState(true);
+  const [selectChain, setSelectChain] = useState(43114);
+
+  const { data: balances, isLoading } = useGetChainsBalances();
+  const { data: chainTokens, isLoading: isLoadingTokens } = useGetWalletChainTokens(selectChain);
+
+  useEffect(() => {
+    if (balances) {
+      const _availableBalances = balances.chains
+        .filter((chain) => chain.balance > 0.01)
+        .sort((a, b) => b.balance - a.balance);
+
+      if (_availableBalances.length > 0) {
+        setSelectChain(_availableBalances[0].chainID);
+      }
+
+      setAvailableBalances(_availableBalances);
+    }
```

Comment:
same code we are repeating in `src/components/Portfolio/Portfolio.tsx` except lines 32 - 34.
sort and filter should be done in `useGetChainsBalances` hook